### DISCUSSION TITLE
How to get Ketcher to open from a smiles code?

### DISCUSSION THREAD

**Discussion by Marco-Matlock** [epam/ketcher#6193] [Q&A] [answered]
Essentially, I have integrated Ketcher into a task pane in Microsoft Excel. The next step for me is to get the active cell smiles code and have Ketcher load that molecule so that I can edit it and then put it back into the active cell in reference.

I opened up the dev console on both Microsoft Edge and Firefox and attempted this line:
ketcher.setMolecule(CCCC) just to see if it would load up CCCC as a smile string but it gives me an error that "Uncaught ReferenceError: CCC is not defined
    at <anonymous>:1:21"

Any ideas without having to actually having to mess within Ketcher's code itself? Thanks.

![image](https://github.com/user-attachments/assets/545ca0f5-d801-417d-9211-c9d3e0209e71)

**Comment by rrodionov91** [comment, marked as answer]
Hi @MarcoMatlock 
I think ketcher.setMolecule('CCCC') should work :)
So please try to pass a string to ketcher.setMolecule instead of variable CCCC which is not defined as I can see.

**Reply by Marco-Matlock** [reply]
Yeah, my mistake was for whatever reason, it prefers everything lowercase. 

ketcher.setMolecule('cccc') was accepted lol.

**Comment by rrodionov91** [comment]
By the way, could you please record small video how the integration with Excel works? Maybe we can include your project at least into our documentation, so it will be useful for other users.
